### PR TITLE
disable RBAC post start hook

### PR DESCRIPTION
**Pull request by deads2k**
pulls in @sttts commit for disabling the hook. Switches @enj's test to use system:masters.  Fixes @smarterclayton's route strategy check to respect groups and tokens

### CODE REVIEW

**Comment by deads2k** [comment]
[test]

File: pkg/route/registry/route/strategy.go [hunk 1]
```diff
@@ -169,14 +171,16 @@ func (s routeStrategy) validateHostUpdate(ctx apirequest.Context, route, older *
 	}
 	res, err := s.sarClient.CreateSubjectAccessReview(
 		ctx,
-		&authorizationapi.SubjectAccessReview{
-			User: user.GetName(),
-			Action: authorizationapi.Action{
-				Verb:     "update",
-				Group:    "route.openshift.io",
-				Resource: "routes/custom-host",
+		authorizationapi.AddUserToSAR(
```
Comment (deads2k) [inline pkg/route/registry/route/strategy.go]: @smarterclayton there's a helper for this

File: test/integration/etcd_storage_path_test.go [hunk 2]
```diff
@@ -851,10 +851,12 @@ func testEtcdStoragePath(t *testing.T, etcdServer *etcdtest.EtcdTestServer, gett
 	}
 	masterConfig.EtcdClientInfo.URLs[0] = testutil.GetEtcdURL()
 
-	kubeConfigFile, err := testserver.StartConfiguredMasterAPI(masterConfig)
+	_, err = testserver.StartConfiguredMasterAPI(masterConfig)
 	if err != nil {
 		t.Fatalf("error starting server: %#v", err)
 	}
+	kubeConfigFile := masterConfig.MasterClients.OpenShiftLoopbackKubeConfig
```
Comment (enj) [inline test/integration/etcd_storage_path_test.go]: This needs a comment about how `system:masters` is special and in sync between kube and origin RBAC.

**Comment by deads2k** [comment]
[merge]

File: pkg/authorization/util/util.go [hunk 3]
```diff
@@ -14,6 +14,8 @@ import (
 // It returns the modified SubjectAccessReview.
 func AddUserToSAR(user user.Info, sar *authorization.SubjectAccessReview) *authorization.SubjectAccessReview {
 	sar.Spec.User = user.GetName()
+	// reminiscent of the bad old days of C.  Copies copy the min number of elements of both source and dest
+	sar.Spec.Groups = make([]string, len(user.GetGroups()), len(user.GetGroups()))
```
Comment (deads2k) [inline pkg/authorization/util/util.go]: @jim-minter @bparees one to catch in future pulls.  This leads to annoying bugs.

Comment (bparees) [inline pkg/authorization/util/util.go]: thanks.

Comment (jim-minter) [inline pkg/authorization/util/util.go]: @deads2k ooh, I'm very sorry I got that wrong. Thanks. I wonder what the rationale was for that particular golang sharp edge.

Comment (deads2k) [inline pkg/authorization/util/util.go]: > @deads2k ooh, I'm very sorry I got that wrong. Thanks. I wonder what the rationale was for that particular golang sharp edge.

They may need to extend the array.  It's like in C where you can try a realloc, but if there's not enough space, you'll need an entirely new memory block.  That results in needing to reassign and golang can't do that without a pointer to it.  Same reason you foo = append(foo, blah).

Also, golang is stuck in 70s.  Blech.

File: pkg/route/registry/route/strategy.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/route/registry/route/strategy.go]: Now I know.

**Comment by deads2k** [comment]
[merge][severity: blocker]

**Pull request merged**